src/loaddb/load_db.c

```diff
@@ -1229,6 +1279,12 @@ load_object_file (load_args * args)
 	}
     }
 
+  if (!args->table_name.empty () && !load_has_authorization (args->table_name, AU_INSERT))
+    {
+      // user not authorized to insert in class
+      return ER_AU_NO_AUTHORIZATION;
```

Comment:
load_has_authorization may encounter other errors too (e.g. class does not exist). the right error should be propagated.